src/components/loading_strategy.ts

```diff
@@ -36,93 +31,63 @@ function defaultOnPlatformApiError(error: Error) {
   console.log(`Error making SSO request: ${error}`)
 }
 
-export function isLoadingWithUrl(props: WidgetLoadingProps): props is UrlLoadingProps {
+function isLoadingWithUrl(props: WidgetLoadingProps): props is UrlLoadingProps {
   return "url" in props
 }
 
-export function isLoadingWithClientProxy(props: WidgetLoadingProps): props is ClientProxyLoadingProps {
+function isLoadingWithClientProxy(props: WidgetLoadingProps): props is ClientProxyLoadingProps {
   return "proxy" in props
 }
 
-export function isLoadingWithPlatformApiSso(props: WidgetLoadingProps): props is PlatformApiLoadingProps {
+function isLoadingWithPlatformApiSso(props: WidgetLoadingProps): props is PlatformApiLoadingProps {
   return "clientId" in props &&
     "apiKey" in props &&
     "userGuid" in props &&
     "environment" in props
 }
 
-export function isLoadingWithBadProps(): never {
+function isLoadingWithBadProps(): never {
   throw new Error(badPropsMessage)
 }
 
-export function useClientProxy<Options>({
-  proxy,
-  widgetType,
-  options,
-  uiMessageWebviewUrlScheme,
-  language,
-  buildProxyRequest = (req) => req,
-  onProxyRequestError = defaultOnClientProxyRequestError,
-}: LoadingParams<ClientProxyLoadingProps, Options>) {
-  const [widgetUrl, setWidgetUrl] = useState<string | null>(null)
-  const req = genClientProxyRequest({ proxy, uiMessageWebviewUrlScheme, language, widgetType, options })
-
-  useEffect(() => {
-    makeClientProxyRequest(buildProxyRequest(req))
-      .then((json) => setWidgetUrl(json.widget_url.url))
-      .catch(onProxyRequestError)
-  }, [])
-
-  return widgetUrl
-}
-
-export function usePlatformApiSso<Options>({
-  apiKey,
-  clientId,
-  userGuid,
-  environment,
-  widgetType,
-  options,
-  uiMessageWebviewUrlScheme,
-  language,
-  onSsoError = defaultOnPlatformApiError,
-}: LoadingParams<PlatformApiLoadingProps, Options>) {
+export function useWidgetUrl<Props extends WidgetLoadingProps & WidgetOptionProps, Opts>(
+  widgetType: Type,
+  props: Props,
+  optsFromProps: (ps: Props) => Opts,
+): string | null {
   const [widgetUrl, setWidgetUrl] = useState<string | null>(null)
 
-  const params = buildPlatformApiRequestParams<Options>(apiKey, clientId, userGuid,
-    environment, widgetType, uiMessageWebviewUrlScheme, language, options)
-
   useEffect(() => {
-    makePlatformApiRequest(params)
-      .then((response) => setWidgetUrl(response.widget_url.url))
-      .catch((error) => onSsoError(error))
+    if (isLoadingWithUrl(props)) {
+      setWidgetUrl(props.url)
+    } else if (isLoadingWithClientProxy(props)) {
+      const { proxy, language } = props
```

Comment:
I'm not a fan of how the proxy and Platform API URL loading code turned out. I have some ideas on how to refactor it so it's cleaner. Once that's cleaned up, this code will be cleaned up as well.